src/manifest.c

```diff
@@ -554,6 +564,13 @@ void subtract_manifests(struct manifest *m1, struct manifest *m2)
 		}
 		list2 = g_list_next(list2);
 	}
+
+	/* now recurse into the included manifests and subtract those as well */
+	while (includes) {
+		mi = includes->data;
+		includes = g_list_next(includes);
+		subtract_manifests(m1, mi, includes);
```

Comment:
After the recursive calls return, "includes" will again be non-NULL, so I'm seeing lot's of duplicate subtractions happening. Can we avoid the duplicates?